server/runtime/src/main/java/org/infinispan/server/configuration/hotrod/HotRodServerConfigurationParser.java

```diff
@@ -123,6 +128,50 @@ private void parseHotRodConnector(XMLExtendedStreamReader reader, ConfigurationB
             }
          }
       }
+      if (implicitAuthentication) {
+         ServerSecurityRealm securityRealm = serverBuilder.endpoint().securityRealm();
+         builder.authentication().enable().securityRealm(securityRealm.getName());
+         String serverPrincipal = null;
+         for (KerberosSecurityFactoryConfiguration identity : securityRealm.getServerIdentities().kerberosConfigurations()) {
+            if (identity.getPrincipal().startsWith("hotrod/")) {
+               builder.authentication()
+                     .addMechanisms(SaslMechanismInformation.Names.GS2_KRB5, SaslMechanismInformation.Names.GSSAPI);
+               serverPrincipal = identity.getPrincipal();
+               break;
+            }
+            Server.log.debugf("Enabled Kerberos mechanisms for Hot Rod using principal '%s'", identity.getPrincipal());
+         }
+         if (securityRealm.hasFeature(ServerSecurityRealm.Feature.TOKEN)) {
+            builder.authentication().addMechanisms(SaslMechanismInformation.Names.OAUTHBEARER);
+            Server.log.debug("Enabled OAUTHBEARER mechanism for Hot Rod");
+         }
+         if (securityRealm.hasFeature(ServerSecurityRealm.Feature.TRUST)) {
```

Comment:
EDIT: We actually don't parse JSON server files because nobody needs it currently, we are mostly rendering the server config via REST. In any case, it'd be nice that the builder themselves take care of enriching the config based on certain circumstances

Reply:
@gustavonalle I've reworked it so that this is now done in the builders. Much cleaner